app/src/main/java/org/fossasia/openevent/app/core/share/ShareEventFragment.java

```diff
@@ -0,0 +1,153 @@
+    private long eventId;
+
+    public static ShareEventFragment newInstance(long eventId) {
+        ShareEventFragment fragment = new ShareEventFragment();
+        Bundle args = new Bundle();
+        args.putLong(MainActivity.EVENT_KEY, eventId);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        final Context contextThemeWrapper = new ContextThemeWrapper(getActivity(), R.style.AppTheme);
+        LayoutInflater localInflater = inflater.cloneInContext(contextThemeWrapper);
+        binding =  DataBindingUtil.inflate(localInflater, R.layout.share_event_layout, container, false);
+        shareEventViewModel = ViewModelProviders.of(this, viewModelFactory).get(ShareEventViewModel.class);
+
+        eventId = getArguments().getLong(MainActivity.EVENT_KEY);
+
+        return binding.getRoot();
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        setupRefreshListener();
+
+        shareEventViewModel.getProgress().observe(this, this::showProgress);
+        shareEventViewModel.getError().observe(this, this::showError);
+        loadEvent(false);
+
+        binding.moreLink.setOnClickListener(view -> shareEvent());
+        binding.emailLink.setOnClickListener(view -> shareByEmail());
+        binding.copyLink.setOnClickListener(view -> copyUrlToClipboard());
+    }
+
+    public void shareEvent() {
+        Intent shareIntent = new Intent();
+        shareIntent.setAction(Intent.ACTION_SEND);
+        shareIntent.putExtra(Intent.EXTRA_TEXT, shareEventViewModel.getShareableInformation());
+        shareIntent.setType("text/plain");
+        startActivity(Intent.createChooser(shareIntent, getResources().getText(R.string.send_to)));
+    }
+
+    public void shareByEmail() {
+        Intent shareIntent = new Intent();
+        shareIntent.setAction(Intent.ACTION_SENDTO);
+        shareIntent.setType("message/rfc822");
+        shareIntent.setData(Uri.parse("mailto:"));
+        shareIntent.putExtra(Intent.EXTRA_SUBJECT, shareEventViewModel.getEmailSubject());
+        shareIntent.putExtra(Intent.EXTRA_TEXT, shareEventViewModel.getShareableInformation());
+        try {
+            startActivity(Intent.createChooser(shareIntent, getResources().getText(R.string.send_to)));
+        } catch (android.content.ActivityNotFoundException ex) {
+            Toast.makeText(getActivity(), "There are no email clients installed.", Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public void copyUrlToClipboard() {
+        ClipboardManager clipboard = (ClipboardManager) getActivity().getSystemService(Context.CLIPBOARD_SERVICE);
+        String eventUrl = shareEventViewModel.getShareableUrl();
+        if (eventUrl == null) {
+            Toast.makeText(getActivity(), "Event does not have a Public URL", Toast.LENGTH_SHORT).show();
```

Comment:
I guess we can use Snackbar so as to maintain consistency as it has been used throughout the app.